os/fs/vfs/fs_stat.c

```diff
@@ -241,3 +245,81 @@ int stat(FAR const char *path, FAR struct stat *buf)
 	set_errno(ret);
 	return ERROR;
 }
+
+/****************************************************************************
+ * Name: inode_stat
+ *
+ * Description:
+ *   The inode_stat() function will obtain information about an 'inode' in
+ *   the pseudo file system and will write it to the area pointed to by 'buf'.
+ *
+ *   The 'buf' argument is a pointer to a stat structure, as defined in
+ *   <sys/stat.h>, into which information is placed concerning the file.
+ *
+ * Input Parameters:
+ *   inode - The inode of interest
+ *   buf   - The caller provide location in which to return information about
+ *           the inode.
+ *
+ * Returned Value:
+ *   Zero (OK) returned on success.  Otherwise, a negated errno value is
+ *   returned to indicate the nature of the failure.
+ *
+ ****************************************************************************/
+
+int inode_stat(FAR struct inode *inode, FAR struct stat *buf)
+{
+	DEBUGASSERT(inode != NULL && buf != NULL);
+
+	/* Most of the stat entries just do not apply */
+
+	RESET_BUF(buf);
```

Comment:
Do we need this #define ? we can directly memset with size of (struct stat) right ?

Reply:
We don't need it right now. Yet if we want to retain code compatibility with nuttx and have easy merging we want to keep it as it is.
Softlinks (if enabled) alter this macro.